examples/seismic/tutorials/09_viscoelastic.ipynb

```diff
@@ -0,0 +1,465 @@
+    "    + damp*s*pi*t_ep/t_s*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - damp*2.*s*mu*t_es/t_s*(vx.forward.dx+vz.forward.dz) \\\n",
+    "            + damp*s*ryy.forward)\n",
+    "# Vertical stress equation:\n",
+    "eq_szz = Eq(szz.forward, damp*szz +\\\n",
+    "    damp*s*pi*t_ep/t_s*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - damp*2.*s*mu*t_es/t_s*(vx.forward.dx+vy.forward.dy) \\\n",
+    "            + damp*s*rzz.forward)\n",
+    "# Diagonal stress equations:\n",
+    "eq_sxy = Eq(sxy.forward, damp*sxy \\\n",
+    "    + damp*s*mu*t_es/t_s*(vx.forward.dy+vy.forward.dx) + damp*s*rxy.forward)\n",
+    "eq_sxz = Eq(sxz.forward, damp*sxz \\\n",
+    "    + damp*s*mu*t_es/t_s*(vx.forward.dz+vz.forward.dx) + damp*s*rxz.forward)\n",
+    "eq_syz = Eq(syz.forward, damp*syz \\\n",
+    "    + damp*s*mu*t_es/t_s*(vy.forward.dz+vz.forward.dy) + damp*s*ryz.forward)\n",
+    "\n",
+    "# Horizontal memory variable equations:\n",
+    "eq_rxx = Eq(rxx.forward, damp*rxx \\\n",
+    "    - damp*s*1./t_s*(rxx+pi*(t_ep/t_s-1)*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - 2*mu*(t_es/t_s-1)*(vz.forward.dz+vy.forward.dy)))\n",
+    "eq_ryy = Eq(ryy.forward, damp*ryy \\\n",
+    "    - damp*s*1./t_s*(ryy+pi*(t_ep/t_s-1)*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - 2*mu*(t_es/t_s-1)*(vx.forward.dx+vy.forward.dy)))\n",
+    "# Vertical memory variable equation:\n",
+    "eq_rzz = Eq(rzz.forward, damp*rzz \\\n",
+    "    - damp*s*1./t_s*(rzz+pi*(t_ep/t_s-1)*(vx.forward.dx+vy.forward.dy+vz.forward.dz) \\\n",
+    "        - 2*mu*(t_es/t_s-1)*(vx.forward.dx+vy.forward.dy)))\n",
+    "# Diagonal memory variable equations:\n",
+    "eq_rxy = Eq(rxy.forward, damp*rxy \\\n",
+    "    - damp*s*1/t_s*(rxy+mu*(t_es/t_s-1)*(vx.forward.dy+vy.forward.dx)))\n",
+    "eq_rxz = Eq(rxz.forward, damp*rxz \\\n",
+    "    - damp*s*1/t_s*(rxz+mu*(t_es/t_s-1)*(vx.forward.dz+vz.forward.dx)))\n",
+    "eq_ryz = Eq(ryz.forward, ryz \\\n",
+    "    - damp*s*1/t_s*(ryz+mu*(t_es/t_s-1)*(vy.forward.dz+vz.forward.dy)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now create and then run the operator:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create the operator:\n",
+    "op = Operator([eq_vx, eq_vy, eq_vz, eq_rxx, eq_ryy, eq_rzz, eq_rxy, eq_rxz, eq_ryz,\n",
+    "               eq_sxx, eq_syy, eq_szz, eq_sxy, eq_sxz, eq_syz] + src_xx + src_yy + src_zz,\n",
+    "               subs=model.spacing_map)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Operator `Kernel` run in 67.81 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Execute the operator. NOTE: This may take up to several minutes.\n",
```

Comment:
Maybe not necessary? With no optimizations no openmp/mpi I got 115 secs on a very weak laptop.